src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -86,19 +97,31 @@ private DiffParser() {
      * @return the tree parser
      * @throws IOException JGit library exception
      */
+    @SuppressWarnings("TryFinallyCanBeTryWithResources")
     private static AbstractTreeIterator prepareTreeParser(Repository repository, String ref)
             throws IOException {
-        final Ref head = repository.exactRef(ref);
+        final CanonicalTreeParser returnValue;
+
         final RevWalk walk = new RevWalk(repository);
-        final RevCommit commit = walk.parseCommit(head.getObjectId());
-        final RevTree tree = walk.parseTree(commit.getTree().getId());
-        final CanonicalTreeParser treeParser = new CanonicalTreeParser();
-        final ObjectReader reader = repository.newObjectReader();
-        treeParser.reset(reader, tree.getId());
-        reader.close();
-        walk.dispose();
-        walk.close();
-        return treeParser;
+        try {
+            final Ref head = repository.exactRef(ref);
+            final RevCommit commit = walk.parseCommit(head.getObjectId());
+            final RevTree tree = walk.parseTree(commit.getTree().getId());
+            returnValue = new CanonicalTreeParser();
```

Comment:
even this doesn't throw an exception, I feel it should still be inside the try.
If that causes trouble, than attach it to `returnValue`'s declaration, or at the very least move it down one line closer to try.